C-Plus-Plus/ds/max_min_heap.cpp

```diff
@@ -0,0 +1,123 @@
+#include <iostream>
+#define max 20
+using namespace std;
+
+class heap{
+    int arr[max+1];
+    int n;
+    public:
+    heap(){
+        cout<<"Enter the total no. of elements in a heap u want ?"<<endl;
+        cin>>n;
+        arr[n+1];
+        for(int i =0;i< n+1;i++){
+            arr[i] = 0;
+        }
+        
+    }
+    void max_heap(){
+        char ch;
+        int j = n;
+        int m ,i=1;
+        do{
+            cout<<"Enter element "<<i<<" :";
+            cin>>m;
+            arr[i] = m;
+            insert_max(i);
+            i++;
+            j--;
+        }while(j>=1);
+        disp();
+    }
+    void insert_max(int i){
+        int elem = arr[i];
+        while(i>1 && arr[i/2]<elem){
+            arr[i] = arr[i/2];
+            i = i/2;
+        }
+        arr[i] = elem;
+    }
+    void min_heap(){
+        char ch;
+        int j = n;
+        int m ,i=1;
+        do{
+            cout<<"Enter element "<<i<<" :";
+            cin>>m;
+            arr[i] = m;
+            insert_min(i);
+            i++;
+            j--;
+        }while(j>=1);
+        disp();
+    }
+    void insert_min(int i){
+        int elem = arr[i];
+        while(i>1 && arr[i/2]>elem){
+            arr[i] = arr[i/2];
+            i = i/2;
+        }
+        arr[i] = elem;
+    }
+    void disp(){
+        for(int i=1;i<n+1;i++){
+            cout<<arr[i]<<"\t";
+        }
+        cout<<"\n";
+    }
+};
+int main()
+{
+    heap h;
+    int ch;
+    do{
+        cout<<"Enter the choice below :\n1.MAX heap \n2.MIN heap \n3.EXIT\n";
+        cin>>ch;
+        switch(ch)
+        {
+            case 1:
+            h.max_heap();
```

Comment:
Display successful message after each operation performed for better User friendly interface.